float32[3] torques